test/FormControlSpec.tsx

```diff
@@ -0,0 +1,157 @@
+import * as React from 'react';
+import { render } from '@testing-library/react';
+
+import FormControl from '../src/FormControl';
+import FormGroup from '../src/FormGroup';
+
+import { shouldWarn } from './helpers';
+
+describe('<FormControl>', () => {
+  it('should render correctly', () => {
+    const { getByTestId } = render(
+      <FormControl
+        type="text"
+        id="foo"
+        name="bar"
+        className="my-control"
+        data-testid="test-id"
+      />,
+    );
+
+    const element = getByTestId('test-id');
+    element.tagName.toLowerCase().should.equal('input');
+    element.id.should.equal('foo');
+    element.classList.length.should.equal(2);
+    element.classList.contains('form-control').should.be.true;
+    element.classList.contains('my-control').should.be.true;
+    element.hasAttribute('name').should.be.true;
+    element.getAttribute('name').should.equal('bar');
+    element.hasAttribute('type').should.be.true;
+    element.getAttribute('type').should.equal('text');
+  });
+
+  it('should support textarea', () => {
+    const { getByTestId } = render(
+      <FormControl as="textarea" data-testid="test-id" />,
+    );
+
+    getByTestId('test-id').tagName.toLowerCase().should.equal('textarea');
+  });
+
+  it('should support plaintext inputs', () => {
+    const { getByTestId } = render(
+      <FormControl plaintext data-testid="test-id" />,
+    );
+
+    const element = getByTestId('test-id');
+    element.classList.length.should.equal(1);
+    element.classList.contains('form-control-plaintext').should.be.true;
+  });
+
+  it('should support type=color', () => {
+    const { getByTestId } = render(
+      <FormControl type="color" data-testid="test-id" />,
+    );
+
+    const element = getByTestId('test-id');
+    element.hasAttribute('type').should.be.true;
+    element.getAttribute('type').should.equal('color');
```

Comment:
Don't need hasAttribute since the value is checked on next line anyway

Reply:
Fixed in the commit https://github.com/react-bootstrap/react-bootstrap/pull/6224/commits/023f9b3e2cffbb507001f992d76bb9cfb4243310